### PR TITLE
fix(ui): Display correct track error message when using electron app

### LINKED ISSUE #224: Prevent "use desktop/electron app" error message ...when using that app!
**Describe the bug**

A few years ago, Google Chrome prevented videos from being played without user intervention, when the tab is not focused.

In order to not let new users think that Openwhyd is not able to play tracks sequentially because of that new behavior, I decided to pause the playback whenever the user would switch to another track, so that they can see the following recommendation:

> Oops, we could not play this track... Please try with Openwhyd Desktop App 👌

Indeed, the "Openwhyd Desktop App" (downloadable from https://openwhyd.org/download) is able to play tracks sequentially, even in the background.

Unfortunately, that same error message is displayed every time a track fails to be played, for any reason, including from the Openwhyd Desktop App itself! 😬

**To Reproduce**

1. Start the Openwhyd Desktop App (you can download the latest versionfrom https://openwhyd.org/download)
2. Open the following posted track from the App: https://openwhyd.org/c/53035251200c8a63270004b3
3. Play the track, it should display the error message

**Expected behavior**

When a track fails to play from the Desktop App, it should displayed another error message. Something like:

> Oops, we could not play this track...

But, when failing to play that track from openwhyd.org (in Google Chrome), the error message should be left as it currently is. (i.e. invite the user to install the Desktop App)

**Screenshots**

Here's what the error looks like:

Capture d’écran 2020-03-15 à 11 10 04

You can find the implementation of that error message in `whydPlayer.js`

### PR DESCRIPTION
**Pull request by compiuta**
Closes #224 .

## What does this PR do / solve?

Show track error message in electron desktop app without "... Please try with Openwhyd Desktop App"

## Overview of changes

I used a suggested solution from this issue https://github.com/electron/electron/issues/2288#issuecomment-337858978

By detecting if the user agent contains "electron" it is possible to show a unique message for desktop browsers or the electron app.

## How to test this PR?

I have posted some issues I am having with testing these changes here https://github.com/openwhyd/openwhyd/issues/224#issuecomment-601430390

### CODE REVIEW

**Comment by adrienjoly** [comment]
Awesome, thank you! I’m working all day, but hope to review this PR before the end of the weekend.

File: public/js/whydPlayer.js [hunk 1]
```diff
@@ -496,19 +496,31 @@ function WhydPlayer() {
...
-    onError: function(e) {
+    onError: function (e) {
       if (currentTrack.metadata.logData) logTrackPlay();
 
+      // will try to fallback failing tracks to deezer player
+      //Detect user agent for electron specific message
+      var userAgent = navigator.userAgent.toLowerCase();
+      var isElectronApp = false;
+      var failedTrackMessage;
+      if (userAgent.indexOf(' electron/') > -1) {
```
Comment (adrienjoly) [inline public/js/whydPlayer.js]: Can we use the full `openwhyd-electron` name, to make the condition even more specific to our own desktop client?

```suggestion
      if (userAgent.indexOf(' openwhyd-electron/') > -1) {
```

File: public/js/whydPlayer.js [hunk 2]
```diff
@@ -496,19 +496,31 @@ function WhydPlayer() {
...
+        isElectronApp = true;
```
Comment (adrienjoly) [inline public/js/whydPlayer.js]: If I understand well, the `isElectronApp` is not used anywhere. If so, can we remove that variable?

File: public/js/whydPlayer.js [hunk 3]
```diff
@@ -496,19 +496,31 @@ function WhydPlayer() {
...
+      } else {
+        failedTrackMessage = 'Oops, we could not play' +
+          ' <a href="' +
+          currentTrack.metadata.url +
+          ' target="_blank">this track</a>...' +
+          ' Please try with <a href="https://openwhyd.org/download"' +
+          ' target="_blank">Openwhyd Desktop App</a> 👌'
+      }
+
       // will try to fallback failing tracks to deezer player
```
Comment (adrienjoly) [inline public/js/whydPlayer.js]: Also, this is not at all critical, but feel free to remove that old comment that does not correspond to the code anymore.

File: public/js/whydPlayer.js (hunk 1 above)
Comment (compiuta) [reply public/js/whydPlayer.js]: Yeah I think that works if the desktop client for openwhyd does have 'openwhyd-electron' in the user agent. I just used the code mentioned in the stackoverflow post that would work for any electron app I suppose

File: public/js/whydPlayer.js (hunk 2 above)
Comment (compiuta) [reply public/js/whydPlayer.js]: Yeah this variable is not used anywhere else so it can definitely be removed

File: public/js/whydPlayer.js (hunk 3 above)
Comment (compiuta) [reply public/js/whydPlayer.js]: got it :+1:

**Comment by compiuta** [comment]
@adrienjoly I noticed that a USING_ELECTRON variable already exists in the javascript file so I added it into the if condition.

**Pull request merged**